File: wxrtc/src/main/java/com/wx/rtc/socket/SocketListener.java
package com.wx.rtc.socket;

import com.wx.rtc.bean.ResultData;

public interface SocketListener{
    void onError(int errCode, String errMsg);
    void onSocketOpen();
    void onLogin();
    void onLogout(int reason);
    void onEnterRoom(String publishUrl);
    void onExitRoom(int reason);
    void onGetUnpublishUrl(String unpublishUrl);
    void onRemoteUserEnterRoom(String pullUrl, String userId);
    void onRemoteUserLeaveRoom(String userId, int reason);
    void onRecvRoomMsg(String streamId, String cmd, String message);
    void onRecvCallMsg(String streamId, String cmd, String roomId);
    void onResult(ResultData resultData);

    void onRecordStart(String fileName);
    void onRecordEnd(String fileName);
}

File: wxrtc/src/main/java/com/wx/rtc/socket/SocketListener.kt
package com.wx.rtc.socket

import com.wx.rtc.WXRTCDef

internal interface SocketListener {
    fun onError(errCode: Int, errMsg: String)
    fun onSocketOpen()
    fun onLogin()
    fun onLogout(reason: Int)
    fun onEnterRoom(publishUrl: String)
    fun onExitRoom(reason: Int)
    fun onGetUnpublishUrl(unpublishUrl: String)
    fun onRemoteUserEnterRoom(pullUrl: String, userId: String)
    fun onRemoteUserLeaveRoom(userId: String, reason: Int)
    fun onUserVideoAvailable(userId: String, available: Boolean)
    fun onUserAudioAvailable(userId: String, available: Boolean)
    fun onRecvRoomMsg(userId: String, cmd: String, message: String)
    fun onRecvCallMsg(userId: String?, cmd: String, roomId: String?)
    fun onResult(processData: WXRTCDef.ProcessData)

    fun onRecordStart(fileName: String)
    fun onRecordEnd(fileName: String)
}
